lib/puppet-lint/plugins/check_resources/quoted_literal.rb

```diff
@@ -0,0 +1,232 @@
+    'system' => BOOLYESNO
+  },
+  'notify' => {
+    'withpath' => BOOL
+  },
+  'package' => {
+    'provider' => Set[
+      'aix',
+      'appdmg',
+      'apple',
+      'apt',
+      'aptitude',
+      'aptrpm',
+      'blastwave',
+      'dnf',
+      'dnfmodule',
+      'dpkg',
+      'fink',
+      'freebsd',
+      'gem',
+      'hpux',
+      'macports',
+      'nim',
+      'openbsd',
+      'opkg',
+      'pacman',
+      'pip2',
+      'pip3',
+      'pip',
+      'pkg',
+      'pkgdmg',
+      'pkgin',
+      'pkgng',
+      'pkgutil',
+      'portage',
+      'ports',
+      'portupgrade',
+      'puppet_gem',
+      'puppetserver_gem',
+      'rpm',
+      'rug',
+      'sun',
+      'sunfreeware',
+      'tdnf',
+      'up2date',
+      'urpmi',
+      'windows',
+      'yum',
+      'zypper',
+    ],
+    'ensure' => OpenSet['present', 'installed', 'absent', 'purged', 'disabled', 'latest'],
+    'allow_virtual' => BOOLYESNO,
+    'allowcdrom' => BOOL,
+    'configfiles' => Set['keep', 'replace'],
+    'enable_only' => BOOLYESNO,
+    'install_only' => BOOLYESNO,
+    'mark' => Set['hold', 'none'],
+    'reinstall_on_refresh' => BOOL
+  },
+  'resources' => {
+    'purge' => BOOLYESNO,
+    'unless_system_user' => OpenSet['true', 'false']
+  },
+  'schedule' => {
+    'period' => Set['hourly', 'daily', 'weekly', 'monthly', 'never'],
+    'periodmatch' => Set['number', 'distance']
+  },
+  'service' => {
+    'ensure' => Set['stopped', 'false', 'running', 'true'],
+    'enable' => Set['true', 'false', 'manual', 'mask', 'delayed'],
+    'hasrestart' => BOOL,
+    'hasstatus' => BOOL,
+    'provider' => Set[
+      'base',
+      'bsd',
+      'daemontools',
+      'debian',
+      'freebsd',
+      'gentoo',
+      'init',
+      'launchd',
+      'openbsd',
+      'openrc',
+      'openwrt',
+      'rcng',
+      'redhat',
+      'runit',
+      'service',
+      'smf',
+      'src',
+      'systemd',
+      'upstart',
+      'windows',
+    ]
+  },
+  'tidy' => {
+    'backup' => OpenSet['false'],
+    'recurse' => OpenSet['true', 'false', 'inf'],
+    'rmdirs' => BOOLYESNO,
+    'type' => Set['atime', 'mtime', 'ctime']
+  },
+  'user' => {
+    'ensure' => Set['present', 'absent', 'role'],
+    'allowdupe' => BOOLYESNO,
+    'attribute_membership' => Set['inclusive', 'minimum'],
+    'auth_membership' => Set['inclusive', 'minimum'],
+    'expiry' => OpenSet['absent'],
+    'forcelocal' => BOOLYESNO,
+    'key_membership' => Set['inclusive', 'minimum'],
+    'managehome' => BOOLYESNO,
+    'membership' => Set['inclusive', 'minimum'],
+    'profile_membership' => Set['inclusive', 'minimum'],
+    'provider' => Set[
+      'aix',
+      'directoryservice',
+      'hpuxuseradd',
+      'ldap',
+      'openbsd',
+      'pw',
+      'user_role_add',
+      'useradd',
+      'windows_adsi',
+    ],
+    'purge_ssh_keys' => OpenSet['true', 'false'],
```

Comment:
The docs state this should be a true boolean or an array of paths: https://github.com/puppetlabs/puppet/blob/9b3514e6ad2ef37e24e551a055667ce882e6cf47/lib/puppet/type/user.rb#L715-L728

This means it could be `BOOL`, but I wonder how it behaves on code where it is an array of strings.

Reply:
It's an `OpenSet`, which means it accepts `true`, `false`, or anything else which isn't a bareword (including lists).

Reply:
That is, you can stack this patch on the PR and it passes just fine:
```diff
diff --git spec/unit/puppet-lint/plugins/check_resources/quoted_literal_spec.rb spec/unit/puppet-lint/plugins/check_resources/quoted_literal_spec.rb
index b7c2333..be2bd82 100644
--- spec/unit/puppet-lint/plugins/check_resources/quoted_literal_spec.rb
+++ spec/unit/puppet-lint/plugins/check_resources/quoted_literal_spec.rb
@@ -58,6 +58,14 @@ describe 'quoted_literal' do
           expect(problems).to contain_error(msg).on_line(1).in_column(28)
         end
       end
+
+      context 'purge_ssh_keys open set' do
+        let(:code) { "user { 'foo': purge_ssh_keys => [ 'yay', 'something' ] }" }
+
+        it 'does not detect any problems' do
+          expect(problems.size).to eq(0)
+        end
+      end
     end

     context 'file with ensure' do
```